test-refined/shared/src/main/scala/zio/test/refined/NumericInstances.scala

```diff
@@ -0,0 +1,63 @@
+package zio.test.refined
+
+import eu.timepit.refined.api.Refined
+import eu.timepit.refined.internal.WitnessAs
+import eu.timepit.refined.numeric.{Greater, Less}
+import zio.test.Gen
+import zio.test.magnolia.DeriveGen
+
+object numeric extends NumericInstances
+
+trait NumericInstances {
+
+  def intGreaterThan[N](implicit
```

Comment:
Add implicit to add `defs`. This will allow users to do `DeriveGen[Int Refined Greater[N]]` to retrieve `Gen`.